airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/AbstractJdbcCompatibleSourceOperations.kt

```diff
@@ -43,6 +43,15 @@ abstract class AbstractJdbcCompatibleSourceOperations<Datatype> :
             val columnName = queryContext.metaData.getColumnName(i)
             val columnTypeName = queryContext.metaData.getColumnTypeName(i)
             try {
+                // attempt to access the column. this allows us to know if it is null before we do
+                // type-specific
+                // parsing. if it is null, we can move on. while awkward, this seems to be the
+                // agreed upon way of checking for null values with jdbc.
+                queryContext.getObject(i)
+                if (queryContext.wasNull()) {
+                    continue
+                }
```

Comment:
I remember this snippet and I'm surprised to see it being added. Was it removed recently?

Reply:
See line 90 below.
This is the same function but returning a `AirbyteRecordDate` rather than a `JsonNode`

Reply:
To add more context : 
1. rowToJson was being overloaded and was the code path for both record parsing + general SQL queries (e.g. querying oplog statistics, table estimates, etc). 
2. As part of the meta error handlng project to deal with serialization erros, a new method was added for record parsing which enriches the `JsonNode` by adding additional metadata. 
3. Eventually, we should audit uses of `rowToJson` and move those queries to some other mechanism as simple queries don't quite need to go through the whole source operatons flow

The `convertDatabaseRowToAirbyteRecordDate` was tested against Postgres, where due to the driver implementaton the `getObject` function throws an error for some data type parsing errors (see dff between rowToJson implementation is Abstract + postgres source operations). 

This was added in a previous PR for Postgres support. With the recent Kotlin migrations - this was applied to Mssql and MySql as well. Ideally, tests should have caught this regression, but the data types tests use the `rowToJson` method which should probably be migrated to use this method

Reply:
Thanks for the explanation!